### PR TITLE
Convert codebase to ES6

### LINKED ISSUE #1171: Convert codebase to JavaScript
Admittedly, most of the reason I don't come back to maintaining this project is that I started it in CoffeeScript in a time before I knew JavaScript well. If it was in JavaScript I'd probably tackle cleaning up the more pesky bits of this code base. Instead, I've been focused on other projects.

It would be super cool to have the code base be modernized.

### PR DESCRIPTION
**Pull request by lgeiger**
Fixes #1171 

Here is a [meaningful diff](https://github.com/lgeiger/atom-script/compare/c6993e0ccbada82ed0a65d73a679b17c35058598...ES6) after the conversion using [`decaffinate`](https://github.com/decaffeinate/decaffeinate).

### CODE REVIEW

**Comment by nixel2007** [comment]
wow, that's awesome! Thanks a lot!
@rgbkrk

File: lib/runner.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        }
+        case 'Project directory of the script': {
+          cwd = this.getProjectPath();
+          break;
+        }
+        case 'Directory of the script': {
+          const pane = atom.workspace.getActivePaneItem();
+          cwd = (pane && pane.buffer && pane.buffer.file && pane.buffer.file.getParent &&
+                 pane.buffer.file.getParent() && pane.buffer.file.getParent().getPath &&
+                 pane.buffer.file.getParent().getPath()) || '';
```
Comment (lgeiger) [inline lib/runner.js]: Can really everything fail?

Before it was:
```
cwd = atom.workspace.getActivePaneItem()?.buffer?.file?.getParent?().getPath?() or ''
```
That's the one thing ES6 is missing 😄

Comment (rgbkrk) [inline lib/runner.js]: I think people would run this with a completely empty workspace sometimes, amongst other things.

**Comment by rgbkrk** [comment]
This is so incredible, thank you for the complete and utter demolition of the old coffeescript. Not sure why the build is hanging though.

Thank you @lgeiger!

File: lib/runner.js (hunk 1 above)
Comment (nixel2007) [inline lib/runner.js]: Heh, this is my code. It is a workaround for strange atom bug on Windows - opening one file as a directory. I'm not sure if every check is needed, it was a hotfix

**Pull request merged**